derive-input/src/internal/table.rs

```diff
@@ -13,13 +13,16 @@ pub(in crate::internal) fn try_parse(
     column_args: &ColumnArgs<'_>,
     plural_name: syn::Ident,
     unique_indices: Vec<syn::Ident>,
+    hook_on_insert: bool,
+    hook_on_update: bool,
+    hook_on_delete: bool,
 ) -> syn::Result<Table> {
     let rust_struct = crate::internal::rust::table::map_struct(input);
 
     let spacetimedb_table = SpacetimeDBTable::map(table_args);
 
     let (spacetimedb_table, spacetimedsl_table) =
-        SpacetimeDSLTable::try_parse(column_args, spacetimedb_table, plural_name, unique_indices)?;
+        SpacetimeDSLTable::try_parse(column_args, spacetimedb_table, plural_name, unique_indices, hook_on_insert, hook_on_update, hook_on_delete)?;
```

Comment:
rename to after_insert, before_update, before_delete.

Also introduce before_insert, after_update and after_delete hooks and allow `#[dsl(hook(after_insert, before_delete, after_update)]` (array) instead of needing to define `hook()` multiple times.